sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -663,4 +662,105 @@ void testMetricStatusManipulationMSRMTActivationStateFAILBadSecondManipulationFa
         verify(mockManipulations).setComponentActivation(METRIC_HANDLE, ComponentActivation.ON);
         verify(mockManipulations).setMetricStatus(METRIC_HANDLE, MetricCategory.MSRMT, ComponentActivation.FAIL);
     }
+
+    @Test
+    @DisplayName("testMetricStatusManipulationSETActivationStateNOTRDYGood: Set ActivationState "
+            + "of all SET-Metrics to NOTRDY.")
+    void testMetricStatusManipulationSETActivationStateNOTRDYGood() {
+
+        // given
+
+        final ComponentActivation startActivationState = ComponentActivation.OFF;
+        final MetricCategory metricCategory = MetricCategory.SET;
+        final ComponentActivation activationState = ComponentActivation.NOT_RDY;
+        final String metricStateHandle = "metricStateHandle";
+
+        setupMetricStatusManipulationSET(metricCategory, metricStateHandle);
+
+        when(mockManipulations.setComponentActivation(metricStateHandle, startActivationState))
+                .thenReturn(ResponseTypes.Result.RESULT_SUCCESS);
+        when(mockManipulations.setMetricStatus(metricStateHandle, metricCategory, activationState))
+                .thenReturn(ResponseTypes.Result.RESULT_SUCCESS);
+
+        // when
+
+        final boolean result =
+                ManipulationPreconditions.MetricStatusManipulationSETActivationStateNOTRDY.manipulation(injector);
+
+        // then
+
+        assertTrue(result);
+        assertFalse(
+                testRunObserver.isInvalid(),
+                "Test run should not have been invalidated. Reason(s): " + testRunObserver.getReasons());
+        verify(mockManipulations).setComponentActivation(metricStateHandle, startActivationState);
+        verify(mockManipulations).setMetricStatus(metricStateHandle, metricCategory, activationState);
+    }
+
+    @Test
+    @DisplayName("testMetricStatusManipulationSETActivationStateNOTRDYBad: First Manipulation failed.")
+    void testMetricStatusManipulationSETActivationStateNOTRDYBadFirstManipulationFailed() {
+
+        // given
+
+        final ComponentActivation startActivationState = ComponentActivation.OFF;
+        final MetricCategory metricCategory = MetricCategory.SET;
+        final ComponentActivation activationState = ComponentActivation.NOT_RDY;
+        final String metricStateHandle = "metricStateHandle";
+
+        setupMetricStatusManipulationSET(metricCategory, metricStateHandle);
+
+        when(mockManipulations.setComponentActivation(metricStateHandle, startActivationState))
+                .thenReturn(ResponseTypes.Result.RESULT_FAIL);
+        when(mockManipulations.setMetricStatus(metricStateHandle, metricCategory, activationState))
+                .thenReturn(ResponseTypes.Result.RESULT_SUCCESS);
+
+        // when
+
+        final boolean result =
+                ManipulationPreconditions.MetricStatusManipulationSETActivationStateNOTRDY.manipulation(injector);
+
+        // then
+        assertFalse(result);
```

Comment:
You could validate, that the testRunObserver was invalidated.

Reply:
Done.